spec/controllers/users_controller_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+describe UsersController do
+
+	let(:user) { build(:user) }
+
+	context 'RESTful user actions' do
+
+		describe 'GET #new' do
+
+			it 'renders the new user page' do
+				get :new
+				expect(response).to render_template :new
+			end
+
+		end
+
+		describe 'POST #create' do
+
+			specify 'valid submissions create new user' do
+				expect do
+					post :create, user: attributes_for(:user)
+				end.to change(User, :count).by(1)
+			end
+
+			it 'redirects to the profile page' do
+				post :create, user: attributes_for(:user)
+				expect(response).
+				to redirect_to user_profile_path(assigns(:user))
+			end
+
+		end
+
+		context 'Logged in User' do
+
+			before do
+				user.save
+				session[:user_id] = user.id
+			end
+
+			#functionality to be implemented
```

Comment:
Good comment! This seems like a good habit when you know more than yourself will be working with the code